File: lib/models/API_Responses/API_Responses.js
const Responses = {
  DefineResponse({ statusCode = 502, message = null, data = null }) {
    return {
      statusCode,
      message,
      data: data,
    };
  },

  R_200({ message = null, data = null }) {
    return this.DefineResponse({ statusCode: 200, message, data });
  },
  R_204({ message = null, data = null }) {
    return this.DefineResponse({ statusCode: 204, message, data });
  },
  R_400({ message = null, data = null }) {
    return this.DefineResponse({ statusCode: 400, message, data });
  },
  R_404({ message = null, data = null }) {
    return this.DefineResponse({ statusCode: 404, message, data });
  },
};

module.exports = Responses;

File: lib/models/API_Responses/API_Responses.go
package API_Responses

type Response struct {
	StatusCode int                    `json:"statusCode"`
	Message    string                 `json:"message"`
	Data       map[string]interface{} `json:"data"`
}

func DefineResponse(statusCode int, message string, data map[string]interface{}) Response {
	return Response{
		StatusCode: statusCode,
		Message:    message,
		Data:       data,
	}
}

func R_200(message string, data map[string]interface{}) Response {
	return DefineResponse(200, message, data)
}
func R_204(message string, data map[string]interface{}) Response {
	return DefineResponse(204, message, data)
}

func R_400(message string, data map[string]interface{}) Response {
	return DefineResponse(400, message, data)
}
func R_404(message string, data map[string]interface{}) Response {
	return DefineResponse(404, message, data)
}
